code/modules/projectiles/crabber.dm

```diff
@@ -0,0 +1,63 @@
+#if ASS_JAM
+/datum/projectile/claw
+
+	name = "crab claw"
+	icon = 'icons/obj/crabgun.dmi'
+	icon_state = "proj_crab"
+
+	power = 50
+	cost = 10
+	dissipation_rate = 1
+	dissipation_delay = 20
+	ks_ratio = 0.0
+	sname = "claw"
+	shot_sound = 'sound/items/wirecutter.ogg'
+	shot_number = 1
+	hit_ground_chance = 20
+	window_pass = 0
+	damage_type = D_ENERGY
+	tick(var/obj/projectile/P)
+		var/turf/T = get_turf(P)
+		if (istype(T,/turf/simulated/floor))
+			T.icon = 'icons/misc/beach.dmi'
+			T.icon_state = "sand"
+			SPAWN_DBG(10 SECONDS)
+				T.icon = initial(T.icon)
+				T.icon_state = initial(T.icon_state)
+	on_hit(atom/hit)
+		..()
+		var/turf/simulated/T2 = get_turf(hit)
+		if (isturf(T2))
+			T2.icon = 'icons/misc/beach.dmi'
+			T2.icon_state = "sand"
+			SPAWN_DBG(10 SECONDS)
+				T2.icon = initial(T2.icon)
+				T2.icon_state = initial(T2.icon_state)
```

Comment:
```suggestion
		animate(src, time = 10 SECONDS)
		animate(icon_state = initial(T2.icon_state))
		animate(icon = initial(T2.icon))
```
There's a neat trick to do this without SPAWNs! Use a long empty `animate` followed by an instantneous `animate` to change the icon state and icon, that will all happen clientside. But try it out first if it works, I'm not 100% sure if `icon` is valid for animate.

Reply:
![animate test](https://user-images.githubusercontent.com/65260770/82131694-90030c80-979d-11ea-9a16-ccb7301bb926.PNG)
It appears this happens.

Reply:
Oh, sorry, made a typo. It should have been:
```
		animate(T2, time = 10 SECONDS)
		animate(icon_state = initial(T2.icon_state))
		animate(icon = initial(T2.icon))
```
(T2 instead of src)

Reply:
It works now, thanks.